src/main/java/com/eva/logic/commands/AddCommentCommand.java

```diff
@@ -0,0 +1,77 @@
+package com.eva.logic.commands;
+
+import static com.eva.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Set;
+
+import com.eva.commons.core.Messages;
+import com.eva.commons.core.index.Index;
+import com.eva.logic.commands.exceptions.CommandException;
+import com.eva.model.Model;
+import com.eva.model.comment.Comment;
+import com.eva.model.person.Address;
+import com.eva.model.person.Email;
+import com.eva.model.person.Name;
+import com.eva.model.person.Person;
+import com.eva.model.person.Phone;
+import com.eva.model.person.staff.Staff;
+import com.eva.model.person.staff.leave.Leave;
+import com.eva.model.tag.Tag;
+
+public class AddCommentCommand extends CommentCommand {
+    public AddCommentCommand(Index index, CommentCommand.CommentPersonDescriptor commentPersonDescriptor) {
+        super(index, commentPersonDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = createAddEditedPerson(personToEdit, commentPersonDescriptor);
+
+        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPerson));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static Person createAddEditedPerson(Person personToEdit,
+                                                CommentCommand.CommentPersonDescriptor commentPersonDescriptor) {
+        assert personToEdit != null;
```

Comment:
Would it be better to change this to use like the standard `requireNonNull(personToEdit)` as used in all other places?
And perhaps, you could import the whole CommentCommand?